src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1577,9 +1577,7 @@ void enableSSL(String host,
         String tmfDefaultAlgorithm = null;  // Default algorithm (typically X.509) used by the TrustManagerFactory
         SSLHandhsakeState handshakeState = SSLHandhsakeState.SSL_HANDHSAKE_NOT_STARTED;
 
-        boolean isFips = false;
         String trustStoreType = null;
-        String fipsProvider = null;
```

Comment:
Hi @nsidhaye ,
Thank you for the comments.

> In this case as FIPSProvider is non-FIPS it's not fip connection.

`fipsProvider` connection property is only used to load the keys. Users will still have to create the keys using a FIPS-approved provider. Non FIPS provider keystores can be used to **store** those keys and this will not affect FIPS compliant mode. Please, take a look at [this article.](https://www.ibm.com/support/knowledgecenter/en/SSYKE2_7.0.0/com.ibm.java.security.component.70.doc/security-component/progpractices.html)

>  One can test it by FipsEnvTest#isFips

Of course this test will fail if you pass in non-FIPS provider. However, as I mentioned above, non FIPS provider is used to **only store the keys.**

> If you use my SSLProtocol fork and use following code for IBM FIPS.

Let me explain what happens when you don't set `fipsProvider` property. In this case, since you did not specify the provider, the driver will traverse the registered providers and find the matching keystore. So if you set `keystoreType` to `JKS`, the matching provider that contains this Keystore is a non-FIPS provider. 

> Made FIPSProvider optional as IBMFIPS does not like FIPSProvider while calling Keystore.getInstance(...)

`Keystore.getInstance(storeType, "IBMJCEFIPS")` does not work, because `IBMJCEFIPS` does not contain any keystores. **That's why we need to use non-FIPS provider to store the keys.** If the given provider contains the specified `storeType`, `Keystore.getInstance(storeType, provider)` is perfectly valid. See [this page](https://www.ibm.com/support/knowledgecenter/SSYKE2_7.0.0/com.ibm.java.security.component.70.doc/security-component/JceDocs/keystore.html). 

I am definitely open to a discussion over a call. Please email me your contact details at v-ulibra@microsoft.com.

Thanks again for valuable comments.